pkg/services/rpcsrv/server.go

```diff
@@ -2368,6 +2368,11 @@ func (s *Server) traverseIterator(reqParams params.Params) (any, *neorpc.Error)
 		}
 	}
 	session.iteratorsLock.Unlock()
+	/*
+		if len(iVals) == 0 {
+			return nil, neorpc.ErrUnknownIterator
+		}
+	*/
```

Comment:
@AnnaShaleva do you expect something like that? I understand that it changes behavior, but after adding these lines I got errors with clients test in several places:
https://github.com/nspcc-dev/neo-go/blob/b961e18480da491f97496b485afb6848d825d8da/pkg/services/rpcsrv/client_test.go#L394
https://github.com/nspcc-dev/neo-go/blob/b961e18480da491f97496b485afb6848d825d8da/pkg/services/rpcsrv/client_test.go#L1670
so I firstly ask you to look.

Reply:
Not exactly this, but these errors you got from the Client test are valid. I'd expect something like:
```

	var (
		iIDStr = iID.String()
		iVals  []stackitem.Item
                ok bool
	)
	for _, it := range session.iteratorIdentifiers {
		if iIDStr == it.ID {
			iVals = iterator.Values(it.Item, count)
                        ok = true
			break
		}
	}
        if !ok {
                return nil, neorpc.ErrUnknownIterator
        }
```
Because iterator can have zero-len number of values (consider iterator that was traversed up to the last value), and this situation is valid.